src/libraries/Common/tests/System/Security/Cryptography/AlgorithmImplementations/SlhDsa/SlhDsaContractTests.cs

```diff
@@ -197,6 +239,366 @@ public static void Dispose_CallsVirtual(SlhDsaAlgorithm algorithm)
+            from algorithm in SlhDsaTestData.AlgorithmsRaw
+            from pbeParameters in new[]
+            {
+                new PbeParameters(PbeEncryptionAlgorithm.TripleDes3KeyPkcs12, HashAlgorithmName.SHA1, 42),
+                new PbeParameters(PbeEncryptionAlgorithm.Aes128Cbc, HashAlgorithmName.SHA256, 1),
+                new PbeParameters(PbeEncryptionAlgorithm.Aes192Cbc, HashAlgorithmName.SHA384, 5),
+                new PbeParameters(PbeEncryptionAlgorithm.Aes256Cbc, HashAlgorithmName.SHA512, 10),
+            }
+            select new object[] { algorithm, pbeParameters };
+
+        public static bool HasSymmetricEncryption => !OperatingSystem.IsBrowser() && !OperatingSystem.IsWasi();
+
+        [ConditionalTheory(nameof(HasSymmetricEncryption))]
+        [MemberData(nameof(AlgorithmWithPbeParametersData))]
+        public static void ExportEncryptedPkcs8PrivateKey_CallsExportSlhDsaPrivateKey(SlhDsaAlgorithm algorithm, PbeParameters pbeParameters)
+        {
+            Action<SlhDsaTestHelpers.ExportEncryptedPkcs8PrivateKeyCallback> test = export =>
+            {
+                using SlhDsaMockImplementation slhDsa = SlhDsaMockImplementation.Create(algorithm);
+
+                slhDsa.ExportSlhDsaSecretKeyCoreHook = _ => { };
+                slhDsa.AddLengthAssertion();
+                slhDsa.AddFillDestination(1);
+
+                // SlhDsaMockImplementation overrides TryExportPkcs8PrivateKeyCore with a stub. In order to replicate the
+                // non-overridden behavior, we will replace the stub with a call to base.TryExportPkcs8PrivateKeyCore.
+                // We can then assert that base.TryExportPkcs8PrivateKeyCore calls ExportSlhDsaSecretKeyCore as expected.
+                slhDsa.TryExportPkcs8PrivateKeyCoreHook = slhDsa.BaseTryExportPkcs8PrivateKeyCore;
+
+                byte[] exported = export(slhDsa, "password", pbeParameters);
+
+                AssertExtensions.GreaterThan(slhDsa.ExportSlhDsaSecretKeyCoreCallCount, 0);
+                AssertExtensions.GreaterThan(slhDsa.TryExportPkcs8PrivateKeyCoreCallCount, 0);
+
+                SlhDsaTestHelpers.AssertEncryptedPkcs8PrivateKeyContents(pbeParameters, exported);
+            };
+
+            SlhDsaTestHelpers.AssertEncryptedExportPkcs8PrivateKey(test, SlhDsaTestHelpers.GetValidPasswordTypes(pbeParameters));
+        }
+
+        [Theory]
+        [MemberData(nameof(SlhDsaTestData.AlgorithmsData), MemberType = typeof(SlhDsaTestData))]
+        public static void TryExportPkcs8PrivateKey_DestinationTooSmall(SlhDsaAlgorithm algorithm)
+        {
+            using SlhDsaMockImplementation slhDsa = SlhDsaMockImplementation.Create(algorithm);
+
+            // Early heuristic based bailout so no core methods are called
+            AssertExtensions.FalseExpression(slhDsa.TryExportPkcs8PrivateKey([], out int bytesWritten));
+            Assert.Equal(0, bytesWritten);
+        }
```

Comment:
Are there no other larger too small sizes to test?  Seed size + 7 is guaranteed too small.  (2 for octet string, 4 for AlgorithmIdentifier(0.0), and 2 more for the wrapping SEQUENCE means a minimum of 8 bytes of overhead).

Going all the way to caring about the encoding size of the OID will get tricky if new SlhDsaAlgorithm values are added in the future with different OID lengths, but something more than just empty still seems worthwhile.

Really this test is "do we return false before we bother asking if the key is not exportable, or it is a public key only"

Reply:
And, of course, this is SLH-DSA, so when I say seed, I mean secret key.

Reply:
Ah, right.  06 01 00.